src/diagnostics/MOM_spatial_means.F90

```diff
@@ -43,37 +43,40 @@ function global_area_mean(var, G, scale, tmp_scale, unscale)
                                                          !! that converts it back to unscaled (e.g., mks)
                                                          !! units to enable the use of the reproducing sums
   real,                    optional, intent(in)  :: tmp_scale !< A temporary rescaling factor for the variable
-                                                         !! that is reversed in the return value [a A-1 ~> 1]
+                                                         !! that is reversed in the return value [a A-1 ~> 1],
+                                                         !! or [b B-1 ~> 1] if unscale is also present.
   real,                    optional, intent(in)  :: unscale !< A rescaling factor for the variable [a A-1 ~> 1]
                                                          !! that converts it back to unscaled (e.g., mks)
-                                                         !! units to enable the use of the reproducing sums.
+                                                         !! units to enable the use of the reproducing sums, or
+                                                         !! a factor converting between rescaled units if
+                                                         !! tmp_scale is also present [B A-1 ~> b a-1].
                                                          !! Here scale and unscale are synonymous, but unscale
-                                                         !! is preferred and takes precedence if both are present.
+                                                         !! is preferred and takes precedence.
   real :: global_area_mean  ! The mean of the variable in arbitrary unscaled units [a] or scaled units [A ~> a]
+                            ! or [B ~> b], depending on which optional arguments are provided
 
   ! Local variables
-  ! In the following comments, [A] is used to indicate the arbitrary, possibly rescaled units of the
-  ! input array while [a] indicates the unscaled (e.g., mks) units that can be used with the reproducing sums
-  real, dimension(SZI_(G),SZJ_(G)) :: tmpForSumming ! An unscaled cell integral [a m2]
-  real :: scalefac   ! An overall scaling factor for the areas and variable [a m2 A-1 L-2 ~> 1]
-  real :: temp_scale ! A temporary scaling factor [a A-1 ~> 1] or [1]
+  ! In the following comments, [A ~> a] is used to indicate the arbitrary, possibly rescaled units of the
+  ! input array while [a] indicates the unscaled (e.g., mks) units that can be used with the reproducing sums.
+  ! [A ~> a] and [B ~> b] are the same units unless tmp_scale and unscale are both present.
+  real, dimension(SZI_(G),SZJ_(G)) :: tmpForSumming ! An unscaled cell integral [a L2 ~> a m2] or [B L2 ~> b m2]
```

Comment:
`tmpForSumming` can also has a unit of `[A L2 ~> a m2]`, when `scalefac` is `[1]`.

Reply:
Thanks for noting this, I have modified this line as suggested.